common/cache/metricsScopeCache.go

```diff
@@ -26,47 +26,96 @@ import (
 	"bytes"
 	"strconv"
 	"sync"
+	"sync/atomic"
+	"time"
 
 	"github.com/uber/cadence/common/metrics"
 )
 
+const flushBufferedMetricsScopeDuration = 10 * time.Second
+
 type domainMetricsScopeCache struct {
-	sync.RWMutex
 	scopeMap map[string]metrics.Scope
```

Comment:
maybe use `metricsScopeMap` instead of `map[string]metrics.Scope` to be consistent?